cli/utils/api.go

```diff
@@ -23,3 +28,49 @@ func GetAPIClient(cliConfig config.Config) *openapi.APIClient {
 
 	return openapi.NewAPIClient(config)
 }
+
+type ResourceClient struct {
+	Client     http.Client
+	BaseUrl    string
+	BaseHeader http.Header
+}
+
+func GetResourceAPIClient(resourceType string, cliConfig config.Config) ResourceClient {
+	transport := &http.Transport{
+		DialContext: (&net.Dialer{
+			Timeout: time.Second,
+		}).DialContext,
+	}
+
+	client := http.Client{
+		Transport: transport,
+		Timeout:   30 * time.Second,
+	}
+
+	baseUrl := fmt.Sprintf("%s://%s/api/%s", cliConfig.Scheme, cliConfig.Endpoint, resourceType)
+	baseHeader := http.Header{
+		"x-client-id":  []string{analytics.ClientID()},
+		"Content-Type": []string{"text/yaml"},
+	}
+
+	return ResourceClient{
+		Client:     client,
+		BaseUrl:    baseUrl,
+		BaseHeader: baseHeader,
+	}
+}
+
+func (resourceClient ResourceClient) GetRequest(url string, method string, body string) (*http.Request, error) {
```

Comment:
One minor detail that we can think of in the future (sorry for not getting this on previous PRs): the GetRequest naming could sometimes be misleading.

At first glance, GetRequest seems to create an HTTP request with the GET Request instead of creating a golang `http.Request` object that accepts any other method. Should we rename this function to another name, like CreateRequest or BuildRequestObject?